### ISSUE TITLE
Progressive growing of GANs?

### ISSUE THREAD

**Issue by shi-weili** [torchgan/torchgan#101]
I've been using TorchGAN to train the global generator/discriminator of pix2pixHD and got great results. I wonder if it's supported to load the pre-trained global models, add local enhancer layers on top of it, and continue the training? 

If the current trainer doesn't support this, could I have some directions on how to implement it? Thanks!

**Comment by avik-pal** [comment]
Progressive GAN can be supported by the default trainer. But it will not be a very pretty solution.

You could train the global model. Now assuming that you named your models `discriminator` and `generator` you could get those attributes of the trainer. Now add the local enhancer layers as you wish to these, call them `discriminator2` and `generator2`. Now simply replace the values of the original attributes with these new updated values. Next train as usual. And continue like this till u want to add more enhancer layers.

(BTW: Do you have the pix2pixHD code available in a public repo? If yes, I would love to link it in the model-zoo readme)

**Comment by shi-weili** [comment]
@avik-pal Thanks for the notes! I'll try it.

I have an implementation of the essential parts of the global generator/discriminator of pix2pixHD (not including label encoder, instance map, or VGG loss). Currently, it's a private repo with a custom Dataset class. If you'd like to link it, I'll repost the model and loss as a public repo probably later this month.

**Comment by avik-pal** [comment]
@shi-weili were you able to make any progress on this? If you need any help, we can try to figure something out

**Comment by shi-weili** [comment]
Thanks for the check-in @avik-pal ! I'll have my GPU available for the experiment in August (it's currently being used for some more urgent rendering task). Hopefully you'll still be available to help by then. 😄 

(Also: I haven't forgotten to share my model. Will find some time to do so.)

**Comment by avik-pal** [comment]
Yeah sure.
